### DISCUSSION TITLE
What are the main innovations of commonware?

### DISCUSSION THREAD

**Discussion by patrick-ogrady** [commonwarexyz/monorepo#240] [Q&A] [answered]
<img width="1153" alt="image" src="https://github.com/user-attachments/assets/3ee8908d-165d-45d2-854b-e42260aa85ef" />

**Comment by patrick-ogrady** [comment, marked as answer]
Great question. I started Commonware with two core insights in mind:
* the dominant onchain applications of the next 10 years will want to ruthlessly specialize the stack to build a defendable moat/robust community (won't be content deploying on general purpose blockspace)
* there have been a number of algorithmic improvements in core components (p2p, cryptography, consensus, storage, etc) over the last 5-7 years but few of these improvements have been adopted because doing so would require projects to significantly depart from their current approaches (which may also be powering a chain)

Commonware is starting from scratch and building a library (MIT/Apache-2) full of primitives that ambitious developers can use to make amazing onchain experiences. They may look like the blockchains you know today or something totally different.

We aspire to build "common software" (neutral open-source software) that enables great developers to push the idea of onchain forward.

***

How does this compare?

* Most focus on providing a general purpose execution layer or allow you to deploy your own general purpose execution layer but don't expect/aren't designed for applications to tune core mechanisms without forking the entire stack (in L2s this might mean your own fraud proving).
* Commonware "Applications" are designed to be run as a standalone chain (own token, no shared security) but can be bootstrapped using external sources of security (like an AVS). They are not made to run in a "single mode" their entire lifetime and can evolve as a community grows.
* Commonware primitives are made to be modded/swapped out. Commonware isn't a "framework" in that you need to have certain blocks/transactions/etc and you can pick-and-choose which primitives you want to use (as I alluded to, you can mod core components without forking things unrelated).

Hope that helps!

I'll wait to touch more on the unique mechanisms we are releasing (and their performance) until we share more.
TL;DR...

reliable open source software for building your own blockchain that incorporates state-of-the-art techniques and is easy to mod

**Comment by patrick-ogrady** [comment]
<img width="1174" alt="image" src="https://github.com/user-attachments/assets/c44dd2f9-ed02-4a1a-80eb-5a97258c13d9" />

**Comment by patrick-ogrady** [comment]
<img width="1175" alt="image" src="https://github.com/user-attachments/assets/059260e2-85d0-43fa-ae57-e4637ed60290" />
